File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    
    # 전화번호가 다른 전화번호의 접두어인지 판단
    # 접두어인지 어캐 판단하지
    # in ?
    # 시간복잡도 고려해야함. 그냥 for문 돌면 시초날수도
    
    phone_book.sort() # 정렬
    
    for i in range(len(phone_book)-1):
        if phone_book[i+1].startswith(phone_book[i]):
            answer= False


    return answer

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        // HashMap 사용
        HashMap<String, Integer> map = new HashMap<>();
        
        // 1.전화번호부 만들기
        for(int i=0; i<phone_book.length; i++){
            map.put(phone_book[i], 1);
        }
        
        // 2.현재 전화번호의 접두어가 map 에 있는지 체크.
        for(int i=0; i<phone_book.length; i++){
            for(int j=1; j<phone_book[i].length(); j++){
                // 전화번호에서 접두어가 포함되는지 체크: substring
                if(map.containsKey(phone_book[i].substring(0,j))){
                    return false;
                }
            }
            
        }
         
        return answer;
    }
}